File: lib/models/user_profile.dart


class UserProfile {
  String id;
  String name;
  String? email;
  double balance;
  int monthlyCount;

  UserProfile({
    required this.id,
    required this.name,
    this.email,
    this.balance = 0.0,
    this.monthlyCount = 0,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'balance': balance,
      'monthlyCount': monthlyCount,
    };
  }

  factory UserProfile.fromMap(Map<String, dynamic> map) {
    return UserProfile(
      id: map['id'] as String,
      name: map['name'] as String,
      email: map['email'] as String?,
      balance: map['balance'] as double,
      monthlyCount: map['monthlyCount'] as int,
    );
  }
}

File: src/models/UserProfile.ts
export interface UserProfile {
  id: string;
  name: string;
  email?: string;
  balance: number;
  monthlyCount: number;
}

export class UserProfileModel implements UserProfile {
  id: string;
  name: string;
  email?: string;
  balance: number;
  monthlyCount: number;

  constructor({
    id,
    name,
    email,
    balance = 0.0,
    monthlyCount = 0,
  }: UserProfile) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.balance = balance;
    this.monthlyCount = monthlyCount;
  }

  toMap(): Record<string, any> {
    return {
      id: this.id,
      name: this.name,
      email: this.email,
      balance: this.balance,
      monthlyCount: this.monthlyCount,
    };
  }

  static fromMap(map: Record<string, any>): UserProfileModel {
    return new UserProfileModel({
      id: map.id as string,
      name: map.name as string,
      email: map.email as string,
      balance: map.balance as number,
      monthlyCount: map.monthlyCount as number,
    });
  }
}
